### DISCUSSION TITLE
MessageBox in Avalonia

### DISCUSSION THREAD

**Discussion by santhoshkumarmk** [AvaloniaUI/Avalonia#6469] [General]
Hi,

I need a MessageBox controls same as in WPF.  I need decision making yes or no messagebox option.
Could any one help us.

**Comment by rstm-sf** [comment]
Hello!

Perhaps you are interested in this repository? https://github.com/AvaloniaCommunity/MessageBox.Avalonia

**Reply by santhoshkumarmk** [reply]
Hi ,

I have used this reference in my project . But the problem is here i need to how we can read the output from this dialog
when user clicks on Yes or No as like we have the option in WPF MessageBox. Can you help on this ?

**Reply by rstm-sf** [reply]
Sorry, but I did not understand you

> read the output from this dialog when user clicks on Yes or No

_Do we mean to read the result of the user's choice?_

It is necessary to await for the result from `ShowDialog` and then we will have the result of the selection
https://github.com/AvaloniaCommunity/MessageBox.Avalonia/blob/67a3bc24c20cabc4d74e6502f276c7f1aa0c0c2d/src/MessageBox.Avalonia.Example/MainWindow.xaml.cs#L69

_Do we need a choice of Yes or No?_

Then we need to choose `ButtonEnum.YesNo`

https://github.com/AvaloniaCommunity/MessageBox.Avalonia/blob/67a3bc24c20cabc4d74e6502f276c7f1aa0c0c2d/src/MessageBox.Avalonia.Example/MainWindow.xaml.cs#L42

**Reply by santhoshkumarmk** [reply]
Thanks for the information.

I have one more question after await how to read user is pressed Yes or No

**Reply by EtherGhost** [reply]
Maybe this sample project can help you with something? 
https://github.com/kvalitetskontrollMarkaryd/AvaloniaWpfMessageDialogService

**Comment by santhoshkumarmk** [comment]
Thanks i got it its working fine

**Comment by EtherGhost** [comment]
Great. Please mark as answered.
